samples/AzureKeyVault/apiDefinition.swagger.json

```diff
@@ -0,0 +1,900 @@
+            }
+          }
+        },
+        "summary": "List secrets"
+      }
+    },
+    "/secrets/{secret-name}": {
+      "get": {
+        "description": "Get a specified secret from a given key vault.",
+        "operationId": "GetSecret",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "secret-name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "default": "7.0",
+            "in": "query",
+            "name": "api-version",
+            "required": true,
+            "type": "string",
+            "x-ms-visibility": "internal"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "default",
+            "schema": {
+              "properties": {
+                "attributes": {
+                  "description": "attributes",
+                  "properties": {
+                    "created": {
+                      "description": "created",
+                      "format": "int32",
+                      "type": "integer"
+                    },
+                    "enabled": {
+                      "description": "enabled",
+                      "type": "boolean"
+                    },
+                    "recoverylevel": {
+                      "description": "recoverylevel",
+                      "type": "string"
+                    },
+                    "updated": {
+                      "description": "updated",
+                      "format": "int32",
+                      "type": "integer"
+                    }
+                  },
+                  "type": "object"
+                },
+                "id": {
+                  "description": "id",
+                  "type": "string"
+                },
+                "value": {
+                  "description": "value",
+                  "format": "byte",
+                  "title": "",
+                  "type": "string",
+                  "x-ms-visibility": ""
+                }
+              },
+              "type": "object"
+            }
+          }
+        },
+        "summary": "Get secret"
+      },
+      "put": {
+        "description": "Sets a secret in a specified key vault. This operation adds a secret to the Azure Key Vault. If the named secret already exists, Azure Key Vault creates a new version of that secret. This operation requires the secrets/set permission.",
+        "operationId": "SetSecret",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "secret-name",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "default": "7.0",
+            "in": "query",
+            "name": "api-version",
+            "required": true,
+            "type": "string",
+            "x-ms-visibility": "internal"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": false,
```

Comment:
"value" field is required, but the parameter itself is optional. It means that empty payload is a valid request.
Is it expected or the parameter should be also required?

Reply:
Good observation. I made body required as well.